src/bin/genemichaels.rs

```diff
@@ -205,7 +243,71 @@ fn main() {
             Offable::On(_) => true,
         },
     };
-    if args.files.is_empty() {
+    if args.folder {
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
+                eprintln!("\x1B[1;32m  Formatting\x1B[0;22m folder...");
+                let c_dir = current_dir()?;
+                let mut manifest = cargo_manifest::Manifest::from_path(c_dir.join(CARGO_TOML))?;
+
+                // this should help re autobins etc
+                manifest.complete_from_path(&c_dir.join(CARGO_TOML))?;
+                process_cargo_toml(c_dir, manifest, config)?;
+                eprintln!(
+                    "\x1B[1;32m    Finished\x1B[0;22m folder formatting successfully in {:.2}s",
+                    time::Instant::now().duration_since(inst).as_secs_f64()
+                );
+                Result::Ok(())
+            }
+        };
+        match res() {
+            Ok(_) => { },
+            Err(e) => {
+                print_error_text();
+                eprintln!("formatting: {:?}", e);
+                process::exit(1);
+            },
+        };
+    } else if args.project {
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
+                eprintln!("\x1B[1;32m  Formatting\x1B[0;22m workspace...");
+                let mut workspace_cargo_toml = None;
+                let c_dir = current_dir()?;
+                let mut at: Option<&Path> = Some(&c_dir);
+                while let Some(d) = at.take() {
+                    if d.join(CARGO_TOML).exists() &&
+                        cargo_manifest::Manifest::from_path(d.join(CARGO_TOML)).is_ok() {
+                        workspace_cargo_toml = Some(d.join(CARGO_TOML));
+                        break;
+                    }
+                    at = d.parent();
+                }
+                let wct = workspace_cargo_toml.ok_or_else(|| anyhow::anyhow!("No Cargo.toml found!"))?;
+                let mut manifest = cargo_manifest::Manifest::from_path(&wct)?;
+
+                // this should help re autobins etc
+                manifest.complete_from_path(&wct)?;
```

Comment:
Sorry, I was probably confusing things before.  I thought there might be some way to disable the autobin/autobench/etc functionality, but it doesn't look like it.

`from_path` already calls `complete_from_path` AFAICT so this doesn't add anything.

Reply:
Ah, if you read the string and do `from_str` instead of `from_path` it won't do autobins etc.  IMO that would make things simpler since it won't be duplicating the scanning we're doing.

Reply:
I'm somewhat confused here, it does look like `from_str` will ignore some implicit stuff, but don't we want that? Or are you indicating that because we manually check the auto* dirs later there is no point using `from_path`? Are we sure the only implicit stuff we gain from using `from_path` is superseded by us manually later?

I'm not too keen on using the `from_str` method (presumably with a `fs::read_to_string`) over the `from_path` unless there is a clear reason to, might just need to think on this one a little bit more.

Reply:
Yeah reasonable points. 

I think I'm leaning towards leaving it as is for now, if we do run into issues we've both identified this as a potential cause, and we can move to `from_str` if it fixes it. Thanks for your thoughts.